giraffe/src/utils/tooltip.ts

```diff
@@ -7,11 +7,51 @@ import {
   Table,
   DomainLabel,
 } from '../types'
+import {
+  FILL,
+  STACKED_LINE_CUMULATIVE,
+  LINE_COUNT,
+} from '../constants/columnKeys'
 
 import {getDomainDataFromLines} from './lineData'
 
 const isVoid = (x: any) => x === null || x === undefined
 
+const orderDataByValue = (
+  originalOrder: number[],
+  nextOrder: number[],
+  data: Array<any>
+) => {
+  const map = {}
+  originalOrder.forEach((key, i) => (map[key] = data[i]))
+  return nextOrder.map(index => map[index])
+}
+
+const getDataSortOrder = (
+  lineData: LineData,
+  hoveredRowIndices: number[],
+  position: LinePosition
+): number[] => {
+  if (position === 'overlaid') {
+    return hoveredRowIndices
+  }
+  const map = {}
+  const values = Object.keys(lineData).reduce(
```

Comment:
why not use map here?

Reply:
`lineData` has multiple properties, each of them is a nested object with some array inside. I am trying to combine all of those arrays into one long array. reduce is good taking many things and turning them into a single thing, so that was my first instinct here.